vendor/src/github.com/docker/libnetwork/resolver.go

```diff
@@ -47,7 +46,7 @@ const (
 	maxExtDNS       = 3 //max number of external servers to try
 	extIOTimeout    = 4 * time.Second
 	defaultRespSize = 512
-	maxConcurrent   = 50
+	maxConcurrent   = 100
```

Comment:
Yes. it is not a full fix. In 1.11 we added code to restrict the number of concurrent DNS queries from a container to reduce the load on daemon processing to 50 concurrent queries/second. But it seems there are cases such as #22185 that this number seems low. Increasing it is a partial fix while @sanimej is evaluating other options.